services/datamanager/datasync/upload_data_capture_file.go

```diff
@@ -26,23 +28,67 @@ func uploadDataCaptureFile(ctx context.Context, client v1.DataSyncServiceClient,
 		return nil
 	}
 
-	uploadMD := &v1.UploadMetadata{
-		PartId:           partID,
-		ComponentType:    md.GetComponentType(),
-		ComponentName:    md.GetComponentName(),
-		MethodName:       md.GetMethodName(),
-		Type:             md.GetType(),
-		MethodParameters: md.GetMethodParameters(),
-		FileExtension:    md.GetFileExtension(),
-		Tags:             md.GetTags(),
+	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && len(sensorData) > 1 {
+		return errors.New("binary sensor data file with more than one sensor reading is not supported")
 	}
 
-	// If it's a large binary file, we need to upload it in chunks.
-	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && f.Size() > MaxUnaryFileSize {
-		if len(sensorData) > 1 {
-			return errors.New("binary sensor data file with more than one sensor reading is not supported")
+	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && md.GetMethodName() == "GetImages" {
+		timeReq := sensorData[0].GetMetadata().GetTimeRequested()
+		timeRec := sensorData[0].GetMetadata().GetTimeReceived()
+		var getImgsRes pb.GetImagesResponse
+		mp := sensorData[0].GetStruct().AsMap()
+		if err := mapstructure.Decode(mp, &getImgsRes); err != nil {
+			return err
+		}
+
+		for _, img := range getImgsRes.Images {
+			newSensorData := []*v1.SensorData{
+				{
+					Metadata: &v1.SensorMetadata{
+						TimeRequested: timeReq,
+						TimeReceived:  timeRec,
+					},
+					Data: &v1.SensorData_Binary{
+						Binary: img.GetImage(),
+					},
+				},
+			}
+			newUploadMD := &v1.UploadMetadata{
+				PartId:           partID,
+				ComponentType:    md.GetComponentType(),
+				ComponentName:    md.GetComponentName(),
+				MethodName:       md.GetMethodName(),
+				Type:             md.GetType(),
+				MethodParameters: md.GetMethodParameters(),
+				FileExtension:    getFileExtFromImageFormat(img.GetFormat()),
+				Tags:             md.GetTags(),
+			}
+			if err := uploadSensorData(ctx, client, newUploadMD, newSensorData, f.Size()); err != nil {
```

Comment:
[nit] Can return `uploadSensorData()` here directly, since that'll return an error or nil.

Reply:
Because it's in the for loop I thought I had to not be returning unless there's an error.

Reply:
Ah you're right, thanks!